lib/mix/lib/mix/tasks/test.ex

```diff
@@ -148,6 +148,8 @@ defmodule Mix.Tasks.Test do
 
   ## Command line options
 
+    * `--app` - runs test only on specified app(s). Must be first option(s) given.
+      May specify multiple apps with multiple `--app` flags.
```

Comment:
It doesn't need to be the first option given. You can first call `OptionParser.parse!(args, strict: @switches)` and then handle apps accordingly.

Reply:
There is a `:keep` option in OptionParser that would allow the app flag to be given multiple times.

Reply:
I tested this out and it seemed to only remember the first one. The other issue I ran into with handling it via OptionParser is that the entire logic of the test function would need to be wrapped in an `if` statement, but breaking it out as done here and in the `cmd` task results in a cleaner code path.

Reply:
You can break the testing function in two today:

```
def run(args) do
  {args, opts} = OptionParser....
  run(args, opts)
end
```

Then you don't need to wrap it all in an `if`.